apacheconfig/wloader.py

```diff
@@ -0,0 +1,204 @@
+
+        For example::
+
+            ItemNode('\\n  option value').whitespace => "\\n  "
+            ItemNode('option value').whitespace => ""
+            ItemNode('\\n  # comment').whitespace => "\\n  "
+        """
+        return self._whitespace
+
+    @whitespace.setter
+    def whitespace(self, value):
+        """See base class. Operates on preceding whitespace."""
+        self._whitespace = value
+
+    @staticmethod
+    def parse(raw_str, options={}, parser=None):
+        """Factory for :class:`apacheconfig.ItemNode` by parsing data from a
+        config string.
+
+        Args:
+            options (dict): Additional options to pass to the created parser.
+                Ignored if another ``parser`` is supplied.
+            parser (:class:`apacheconfig.ApacheConfigParser`): optional, to
+                re-use an existing parser. If ``None``, creates a new one.
+        Returns:
+            :class:`apacheconfig.ItemNode` containing metadata parsed from
+            ``raw_str``.
+        """
+        if not parser:
+            parser = _create_apache_parser(options, start='startitem')
+        return ItemNode(parser.parse(raw_str))
+
+    @property
+    def name(self):
+        """Returns the name of this node.
+
+        The name is the first non-whitespace token in the directive. Cannot be
+        written. For comments, is the entire comment.
+        """
+        return self._raw[0]
+
+    @property
+    def has_value(self):
+        """Returns ``true`` if this :class:`apacheconfig.ItemNode` has a value.
+
+        ``ItemNode`` objects don't have to have a value, like option/value
+        directives with no value, or comments.
+        """
+        return len(self._raw) > 1
+
+    @property
+    def value(self):
+        """Returns the value of this item as a string.
+
+        The "value" is anything but the name. Can be overwritten.
+        """
+        if not self.has_value:
+            return None
+        return self._raw[-1]
+
+    @value.setter
+    def value(self, value):
+        """Sets for the value of this item.
+
+        Args:
+            value (str): string to set new value to.
+
+          .. todo:: (sydneyli) convert `value` to quotedstring when quoted
+        """
+        if not self.has_value:
+            self._raw = self._raw + (" ", value,)
+        self._raw = self._raw[0:-1] + (value,)
+
+    def dump(self):
+        return (self.whitespace +
+                "".join([_restore_original(word) for word in self._raw]))
+
+    def __str__(self):
+        return ("%s(%s)"
+                % (self.__class__.__name__,
+                   str([self._type] +
+                       [_restore_original(word) for word in self._raw])))
+
+
+def _create_apache_parser(options={}, start='contents'):
+    """Returns a :class:`apacheconfig.ApacheConfigParser` with default options
+    that are expected by Apache's native parser.
+
+    Overrides apacheconfig options ``preservewhitespace``,
```

Comment:
What about `namedblocks` option? Should we also disable it to comply with original Apache syntax?

Reply:
Apache syntax supports named blocks, though they call it something different (arguments rather than block name); for instance, the [Directory](https://httpd.apache.org/docs/2.4/mod/core.html#directory) directive.